### PR TITLE
chore(api-1721): add exception documentation

### PR DESCRIPTION
**Pull request by tfehringer**

### CODE REVIEW

**Comment by LevFlavien** [comment]
Related Client PR https://github.com/akeneo/api-php-client/pull/221

**Review by LevFlavien — COMMENTED** [review]
Since many exceptions have a generic behavior, I suggest we reduce snippet sizes to what is specific for each exception.

eg:

Under main "Client exception" section
=> put the complete snipped

Then, under Forbidden exception
```php
try {
    $client->getProductApi()->get('top');
} catch (ForbiddenHttpException $e) {
    // do your stuff with the exception
}
```

Too many requests exception
```php
try {
    $client->getProductApi()->get('top');
} catch (ForbiddenHttpException $e) {
    // do your stuff with the exception
    $retryAfter = $e->getRetryAfter();
}
```

Or we could even remove all generic snippets and leave only one under "Client exception" section.

**Review by LevFlavien — COMMENTED** [review]
Reviewed together:
- Add HTTP code to titles
- Remove generic snippets, leave only one
- Keep specific snippets for 429 & 422
- Optional : Link to REST doc https://api.akeneo.com/documentation/responses.html#

File: content/php-client/exception.md [hunk 1]
```diff
@@ -126,3 +95,8 @@ try {
     }
 }
 ```
+
+### Too many requests exception (429)
+
+This exception is thrown if too many requests are sent in an amount of time ("rate limiting").
+It corresponds to the HTTP code 429.
```
Comment (LevFlavien) [inline content/php-client/exception.md:102]: On peut ajouter un snippet pour la méthode spécifique https://github.com/akeneo/api-php-client/blob/7bbf258fba18ad86bad699ed7ab6c461cc82b9a7/src/Exception/TooManyRequestsHttpException.php#L14

**Review by dFayet — APPROVED** [review]

**Review by LevFlavien — APPROVED** [review]

**Pull request merged**